src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```diff
@@ -0,0 +1,998 @@
+		}
+		activePresetResult = String.valueOf(presetResponse.get(BarcoE2Constant.NAME).asText());
+		return activePresetResult;
+	}
+
+	/**
+	 * Preset Control: Populate data for preset
+	 *
+	 * @param stats Map of statistics
+	 * @param controls List of AdvancedControllableProperty
+	 * @param dropDownOptions List of dropdown options
+	 */
+	private void populatePresetStatsAndControls(Map<String, String> stats, List<AdvancedControllableProperty> controls, List<String> dropDownOptions, String activePresetResult) {
+		if (dropDownOptions.isEmpty()) {
+			stats.put(BarcoE2ControllingMetric.PRESETS_PRESET.getName(), BarcoE2Constant.NONE);
+		} else {
+			stats.put(BarcoE2ControllingMetric.PRESETS_PRESET.getName(), activePresetResult);
+			controls.add(createDropdown(BarcoE2ControllingMetric.PRESETS_PRESET.getName(), activePresetResult, dropDownOptions));
+		}
+		stats.put(BarcoE2ControllingMetric.PRESETS_LAST_CALLED_PRESET.getName(), activePresetResult);
+		stats.put(BarcoE2ControllingMetric.PRESETS_PRESET_ACTIVATE.getName(), activePresetResult);
+		controls.add(createButton(BarcoE2ControllingMetric.PRESETS_PRESET_ACTIVATE.getName(),
+				BarcoE2Constant.LABEL_ACTIVATE_ON_PROGRAM, BarcoE2Constant.LABEL_PRESSED_RECALLING_PRESET, BarcoE2Constant.GRACE_PERIOD));
+	}
+
+	/**
+	 * Preset Control: Activate the preset
+	 *
+	 * @return Boolean this returns true/false based on the result of activating preset
+	 */
+	private boolean activatePresetResult() {
+		try {
+			boolean result;
+			if (Objects.equals(lastPresetName, BarcoE2Constant.DOUBLE_QUOTES)) {
+				// first time running
+				String activePresetResult = getActivePresetName();
+				if (activePresetResult.equals(BarcoE2Constant.NONE)) {
+					result = activatePreset(lastPresetName);
+				} else {
+					result = activatePreset(activePresetResult);
+				}
+			} else {
+				result = activatePreset(lastPresetName);
+			}
+			return result;
+		} catch (Exception e) {
+			logger.error("Failed to activate preset");
+			throw new CommandFailureException(this.getAddress(), "activatePreset", "Failed to activate preset");
+		}
+	}
+
+	/**
+	 * Preset Control: Activate a preset by name and type
+	 *
+	 * @param presetName name of the preset that need to be activated
+	 * @return boolean type indicates that preset is successfully activated.
+	 */
+	private boolean activatePreset(String presetName) throws Exception {
+		Map<Object, Object> param = new HashMap<>();
+		param.put(BarcoE2Constant.PRESET_NAME, presetName);
+		param.put(BarcoE2Constant.TYPE, 1);
+		JsonNode activatePresetResponse = getByMethod(BarcoE2Constant.METHOD_ACTIVATE_PRESET, param);
+		if (activatePresetResponse == null) {
+			throw new CommandFailureException(this.getAddress(), "activatePreset", "Failed to activate preset");
+		}
+		return activatePresetResponse.get(BarcoE2Constant.SUCCESS_STATUS).asInt() == 0;
+	}
+
+	/**
+	 * Routing control: class to store properties of source
+	 */
+	class SourceProperties {
```

Comment:
please move inner class definitions on top 
General order convention would be: 

- inner classes
- constants/static variables
- static methods
- instance variables
- constructors
- configuration (property getters/setters)
- life cycle methods (init, destroy, etc)
- public interface methods (grouped by interface)
- protected methods
- private methods.

In each group, it is also recommended to keep members sorted alphabetically.

Reply:
Updated.

Reply:
I didn't sort the protected/private methods alphabetically, because I group it based on its function (Device Information, Preset Control, Routing Control).